lib/tests/test_git.rs

```diff
@@ -139,7 +140,12 @@ fn get_git_repo(repo: &Arc<ReadonlyRepo>) -> gix::Repository {
 fn fetch_import_all(mut_repo: &mut MutableRepo, remote: &RemoteName) -> GitImportStats {
     let git_settings = GitSettings::from_settings(mut_repo.base_repo().settings()).unwrap();
     let import_options = default_import_options();
-    let mut fetcher = GitFetch::new(mut_repo, &git_settings, &import_options).unwrap();
+    let mut fetcher = GitFetch::new(
+        mut_repo,
+        GitSubprocessOptions::from_settings(&git_settings),
```

Comment:
nit: it's probably better to add helper function like `subprocess_options(repo.settings())`. Many callers won't need `GitSettings`.

Reply:
i did check the existing callers, and all but one (`jj gerrit upload`) need the GitSettings for other purposes. That could certainly change, though... what I'll do is make GitSubprocessOptions::from_settings load directly from a UserSettings and add to_subprocess_options() for the case where a GitSettings is already loaded.